task2_Gal_Malka/include/ChessBoard.h

```diff
@@ -0,0 +1,40 @@
+#pragma once
+#include "ChessPiece.h"
+#include "Pawn.h"
+#include "Knight.h"
+#include "King.h"
+#include "Queen.h"
+#include "Rook.h"
+#include "Bishop.h"
+
+const int BOARD_SIZE = 8;
+
+enum GameCode {
+	EmptyPieceInSource = 11,
+	SourcePieceWrongColor,
+	DestinationPieceWrongColor,
+	IllegalMovement = 21,
+	CauseCheckmate = 31,
+	LegalAndCheck = 41,
+	Legal
+};
+
+class ChessBoard {
+
+private:
+
+	inline static Color m_turn = Color::White;
+	vector<vector<ChessPiece*>> m_board;
```

Comment:
Since the board has a static known size, I think `std::array` is a better choice for a container then `std::vector`.